pkg/controller/remotewatcher/predicate.go

```diff
@@ -0,0 +1,51 @@
+package remotewatcher
+
+import (
+	migapi "github.com/fusor/mig-controller/pkg/apis/migration/v1alpha1"
+	kapi "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// Correlation labels.
+var labels = map[string]bool{
+	migapi.Label(migapi.MigPlan{}):            true,
+	migapi.Label(migapi.MigCluster{}):         true,
+	migapi.Label(migapi.MigStorage{}):         true,
+	migapi.Label(migapi.MigAssetCollection{}): true,
+	migapi.Label(migapi.MigMigration{}):       true,
+	migapi.Label(migapi.MigStage{}):           true,
+}
+
+type SecretPredicate struct {
+	predicate.Funcs
+}
+
+// Watched resource has been updated.
+func (r SecretPredicate) Update(e event.UpdateEvent) bool {
+	secret, cast := e.ObjectOld.(*kapi.Secret)
+	if cast {
+		return hasCorrelationLabel(secret)
```

Comment:
Filtering so that we only reconcile `Secrets` which we own seems like a really good move here :+1: 

Thinking through the potential benefits of such a predicate for other parts of the system... I don't think we will get the same advantage anywhere that we currently uses `migRef`  to narrow down events that we reconcile based on. 

https://github.com/fusor/mig-controller/blob/master/pkg/controller/migcluster/migcluster_controller.go#L88-L98